lib/Tomcat/ServletTest.pm

```diff
@@ -53,7 +73,16 @@ sub request_information() {
 
 # test request header example
 sub request_header() {
-    assert_screen('tomcat-request-header-example', TIMEOUT);
+    my $mod_jk = shift;
+    if ($mod_jk) {
+        if (is_sle('<=12-sp5')) {
+            assert_screen('tomcat-request-header-example-sle12-jk-mod', TIMEOUT);
+        } else {
+            assert_screen('tomcat-request-header-example-jk-mod', TIMEOUT);
+        }
+    } else {
+        assert_screen('tomcat-request-header-example', TIMEOUT);
+    }
```

Comment:
Maybe I am wrong on this, but I think it will work if you create multiple needles with the same tag
eg 
```suggestion
   assert_screen('tomcat-request-header-example', TIMEOUT);
```

Because openqa will try to find from those needles with the same tag the one with the highest matching, I think you do not need to do any branching at all. Unless it is a peculiar case

Reply:
@ge0r  Thank you so much for the input. It helped and was able to make the changes.